### PR TITLE
Routes setup

### PR DESCRIPTION
**Pull request by jvaneyken**
Screenshot 2023-11-08 at 10 52 40 AM

Screenshot 2023-11-08 at 10 16 53 AM

Screenshot 2023-11-08 at 10 17 12 AM

Screenshot 2023-11-08 at 10 17 27 AM

Screenshot 2023-11-08 at 10 17 58 AM

### CODE REVIEW

File: client/components/ZoopsIndex.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const ZoopsIndex = () => {
```
Comment (paloobi) [inline client/components/ZoopsIndex.tsx:3]: nit: I'm not so sure about this naming?  Seems a little non-standard. I would expect something like ZoopsList, ZoopsView, ZoopsPage - or even just MainView or MainPage

File: client/components/ZoopsShow.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { useParams } from "react-router-dom";
+
+const ZoopsShow = () => {
```
Comment (paloobi) [inline client/components/ZoopsShow.tsx:4]: nit: naming here also feels a little non-standard to me - I would expect something like ZoopDetail or SingleZoopView or SingleZoopPage

Comment (paloobi) [reply client/components/ZoopsShow.tsx:4]: I think the plural in the name is especially throwing me

Comment (paloobi) [reply client/components/ZoopsShow.tsx:4]: FYI I had made suggestions for components and their naming in the Jira (these are all just suggestions though, it's ultimately up to you both to decide on things like naming conventions)

**Review by paloobi — CHANGES_REQUESTED** [review]
RC for:

- no test plan - please include screenshots of all new views added to the app

(will also help with debugging zoop/:id)

File: client/components/ZoopsIndex.tsx (hunk 1 above)
Comment (dyazdani) [reply client/components/ZoopsIndex.tsx:3]: Josh mentioned that this was a naming convention he taught at App Academy. Although it seemed like he was unsure if "index" was the exact word when we were working through it. @jvaneyken Was "index" the word they suggested at App Academy for this sort of page?

File: client/components/ZoopsShow.tsx (hunk 2 above)
Comment (dyazdani) [reply client/components/ZoopsShow.tsx:4]: I'll take a look at those. Agreed that "ZoopsView" is a little confusing cause of the plural.

**Comment by dyazdani** [comment]
Your mention of testing reminds me that I forgot to ask you what sort of testing you would like for the front-end. Should we do Jest testing suites?

File: client/components/ZoopsShow.tsx (hunk 2 above)
Comment (jvaneyken) [reply client/components/ZoopsShow.tsx:4]: Yeah I think the plural there was a mistake. Definitely not married to the naming in any case. Index and Show were just what App Academy taught us

**Comment by jvaneyken** [comment]
@dyazdani those changes look good to me

File: client/components/ZoopList.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { Outlet } from "react-router-dom";
+
+const ZoopsList = () => {
+  return (
+    <>
+      <p>I am the Zoops List</p>
+      
+      <Outlet />
```
Comment (paloobi) [inline client/components/ZoopList.tsx:9]: Why is there an Outlet here? I don't think it's needed, this page shouldn't need a sub-route

File: client/App.tsx [hunk 4]
```diff
@@ -1,7 +1,26 @@
...
 
 const App: React.FC = () => {
-  return <h1>Zoop is cool</h1>;
+  return (
+    <>
+      <Routes>
+        <Route path="/" element={<HomePage />}/>
+        <Route path="/zoops" element={<ZoopsList />}>
+          <Route path=":id" element={<Zoop />} />
+        </Route>
```
Comment (paloobi) [inline client/App.tsx:17]: You shouldn't need to do sub-routing like this? I am pretty sure that React Router v6 supports exact matches. But if you're encountering issues, you could put /zoops/:id first in the list of routes instead

**Review by paloobi — COMMENTED** [review]
I don't think the sub-route + outlet is necessary - that is best saved for cases where you have multiple complex sub-routes nested inside one, it shouldn't be needed in this case where it's just /zoops and /zoops/:id

File: client/App.tsx [hunk 5]
```diff
@@ -1,7 +1,25 @@
...
+import MePage from "./components/MePage";
+import HomePage from "./components/HomePage";
 
 const App: React.FC = () => {
-  return <h1>Zoop is cool</h1>;
+  return (
+    <>
+      <Routes>
+        <Route path="/" element={<HomePage />}/>
+        <Route path="/zoops" element={<ZoopsList />} />
```
Comment (paloobi) [inline client/App.tsx:15]: One last change - the spec calls for / to display a list of all zoops. Intention here was for ZoopsList to be a child component of HomePage, so no need for separate /zoops route

Comment (paloobi) [inline client/App.tsx:15]:
```suggestion
        <Route path="/" element={<HomePage />}/>
```

**Review by paloobi — CHANGES_REQUESTED** [review]
One last change request, to match the specification doc: https://diybootcamp.atlassian.net/wiki/spaces/SD/pages/10616834/Zoop+Social+Media+Platform

**Review by paloobi — APPROVED** [review]

**Pull request merged**